runtime/libcn/src/utils.c

```diff
@@ -428,6 +446,19 @@ void *cn_calloc(size_t num, size_t size)
   }
 }
 
+// TODO - revisit this Dhruv over-confident change *)
+void cn_free_sized(void* malloced_ptr, size_t size)
+{
+  printf("C ownership checking: remove malloc-ed:" FMT_PTR ", size: %lu\n", (uintptr_t) malloced_ptr, size);
+  for (int i = 0; i < size; i++) {
+      signed long *address_key = alloc(sizeof(long));
+      *address_key = (uintptr_t) malloced_ptr + i;
+      /* printf(" off: %d [" FMT_PTR "]\n", i, *address_key); */
+      int curr_depth = ownership_ghost_state_get(address_key);
+      c_ghost_assert(convert_to_cn_bool(curr_depth == cn_stack_depth));
```

Comment:
should the free function itself be asserting anything about ownership? surely that is what the runtime CN spec is for, so this might be stronger than necessary

Reply:
I'll add this as a test.
```c
int *malloc_int()
/*@ trusted; ensures take X = Block<int>(return); @*/
{
   return malloc(sizeof(int));
}

void bad_free(int *p)
/*@ trusted; ensures true; @*/
{
    cn_free_size(p, sizeof(int));
}

int main()
/*@ trusted; @*/
{
    int * p = malloc_int();
    bad(p):
}
```